### DISCUSSION TITLE
Pod settings - which can be used multiple times

### DISCUSSION THREAD

**Discussion by alanbchristie** [nextflow-io/nextflow#2973] [Q&A]
In the documentation for a [Pod](https://www.nextflow.io/docs/latest/process.html#pod) it states...

> _"Some settings, including environment variables, configs, secrets, volume claims, and tolerations, can be specified multiple times for different values."_

As it's not clear precisely what settings can be repeated (because of the word _including_) can the documentation be adjusted to clearly identify which settings can be used multiple times?

My specific interest is "pullSecret" - it is legitimate to provide more than one in a Pod, can this setting be specific more than once?

**Comment by bentsherman** [comment]
You can see here which pod options can be repeated: https://github.com/nextflow-io/nextflow/blob/master/modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy

It looks like the comprehensive list is:
- labels
- annotations
- environment variables
- config maps
- secrets
- volume claims
- tolerations

A pod spec can only have one image pull secret, although you could define a different pull secret for different pods.

Feel free to edit the docs and submit a PR, otherwise I will get around to it.
